packages/neos-ui-editors/src/Library/AssetUpload.js

```diff
@@ -43,6 +45,36 @@ export default class AssetUpload extends PureComponent {
         });
     }
 
+    handleMultiUpload = files => {
+        this.setState({
+            isLoading: true
+        });
+        const index = files.length;
+        const {multipleData} = this.props;
+        const values = multipleData ? multipleData.slice() : [];
+        this.uploadMultipleFiles(index, values, files);
+    }
+
+    uploadMultipleFiles(index, values, files) {
+        index--;
+        const {uploadAsset} = backend.get().endpoints;
+        const {onAfterUpload, focusedNode, siteNodePath} = this.props;
+
+        if (index < 0) {
+            if (onAfterUpload) {
+                onAfterUpload(values);
+            }
+            this.setState({
+                isLoading: false
+            });
+            return;
+        }
+        uploadAsset(files[index], this.props.propertyName, focusedNode, siteNodePath).then(res => {
```

Comment:
this will only work with image assets. For other assets it's missing a metadata parameter.
I'd make it a prop, so it could be used both from Image and Asset editors.